tests/tf-manifests/aws/kms/main.tf

```diff
@@ -15,18 +15,31 @@ provider "aws" {
 data "aws_caller_identity" "current" {
 }
 
-resource "aws_kms_key" "cluster_kms_key" {
-  description = "BYOK Test Key for API automation"
-  tags = {
+locals {
+  path = coalesce(var.path, "/")
+
+  iamRoles = var.hcp ? [
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root",
+    ] : [
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root",
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role${local.path}${var.account_role_prefix}-Installer-Role",
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role${local.path}${var.account_role_prefix}-Support-Role",
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role${local.path}${var.account_role_prefix}-ControlPlane-Role",
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role${local.path}${var.account_role_prefix}-Worker-Role"
+  ]
+
+  defaultTags = {
     Key         = var.tag_key
     Value       = var.tag_value
     Description = var.tag_description
   }
-  deletion_window_in_days = 7
+  optionalTags = var.hcp ? { red-hat = true } : {}
```

Comment:
Is this to match `red-hat-managed` tag or is it something specific to test tags?

Reply:
Isn't this mandatory for KMS key in HCP ?
See first step in https://docs.openshift.com/rosa/rosa_hcp/rosa-hcp-creating-cluster-with-aws-kms-key.html#creating-cluster-with-aws-kms-key